libs/redis/langchain_redis/vectorstores.py

```diff
@@ -892,6 +854,120 @@ def similarity_search(
         embedding = self._embeddings.embed_query(query)
         return self.similarity_search_by_vector(embedding, k, filter, sort_by, **kwargs)
 
+    def prepare_docs(
+        self,
+        return_all,
+        results,
+        return_metadata,
+        with_vectors=False,
+        with_scores=False,
+    ):
+
+        docs = []
+
+        for res in results:
+            if not return_all:
+                metadata = (
+                    {
+                        field.name: res[field.name]
+                        for field in self._index.schema.fields.values()
+                        if field.name
+                        not in [self.config.embedding_field, self.config.content_field]
+                    }
+                    if return_metadata
+                    else {}
+                )
+            else:
+                metadata = {
+                    k: v for k, v in res.items() if k != self.config.content_field
+                }
+
+            lc_doc = Document(
+                page_content=res[self.config.content_field],
+                metadata=metadata,
+            )
+
+            if with_scores:
+                vector_distance = float(res.get("vector_distance", 0))
+                parsed = (lc_doc, vector_distance)
+
+            if with_vectors:
+                vector = self.convert_vector(res)
+                parsed = (lc_doc, vector_distance, vector)
+
+            if not with_scores and not with_vectors:
+                parsed = lc_doc
+
+            docs.append(parsed)
+
+        return docs
+
+    def convert_vector(self, obj: dict):
+        vector = obj.get(self.config.embedding_field)
+        if isinstance(vector, bytes):
+            vector = buffer_to_array(vector, dtype=self.config.vector_datatype)
+        if isinstance(vector, str):
+            vector = ast.literal_eval(vector)
+
+        return vector
+
+    def prepare_docs_full(
```

Comment:
im generally in favor of marking these kinds of methods as "private" since they have no end user value. just add a single underscore `_` to the front. While not a firm gate, it helps when reading source code or docs.